### PR TITLE
Reverted change handling bigint output parameters

### PR DESCRIPTION
**Pull request by david-puglielli**
Reverted #567. Fixes the issue in #699.

This change is [Reviewable](https://reviewable.io/reviews/microsoft/msphpsql/744)

### CODE REVIEW

File: test/functional/pdo_sqlsrv/pdo_bigint_outparam.phpt [hunk 1]
```diff
@@ -26,7 +26,8 @@ insertRow($conn, $tbname, array("c1_bigint" => 922337203685479936));
 $outSql = "{CALL $spname (?)}";
 $bigintOut = 0;
 $stmt = $conn->prepare($outSql);
-$stmt->bindParam(1, $bigintOut, PDO::PARAM_INT, PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE);
+// $stmt->bindParam(1, $bigintOut, PDO::PARAM_INT, PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE);
+$stmt->bindParam(1, $bigintOut, PDO::PARAM_STR, 2048);
 $stmt->execute();
 printf("Large bigint output:\n" );
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_bigint_outparam.phpt]: please remove the commented lines in the test

File: test/functional/sqlsrv/sqlsrv_bigint_outparam.phpt [hunk 2]
```diff
@@ -25,15 +25,19 @@ AE\insertRow($conn, $tbname, array("c1_bigint" => 922337203685479936));
 // Call stored procedure with SQLSRV_PARAM_OUT
 $outSql = "{CALL $spname (?)}";
 $bigintOut = 0;
-$stmt = sqlsrv_prepare($conn, $outSql, array(array(&$bigintOut, SQLSRV_PARAM_OUT)));
+// $stmt = sqlsrv_prepare($conn, $outSql, array(array(&$bigintOut, SQLSRV_PARAM_OUT)));
+// $stmt = sqlsrv_prepare($conn, $outSql, array(array(&$bigintOut, SQLSRV_PARAM_OUT, null, SQLSRV_SQLTYPE_BIGINT))); <-- this works
+$stmt = sqlsrv_prepare($conn, $outSql, array(array(&$bigintOut, SQLSRV_PARAM_OUT, null, SQLSRV_SQLTYPE_VARCHAR(256)))); //<-- this also works
 sqlsrv_execute($stmt);
```
Comment (yitam) [inline test/functional/sqlsrv/sqlsrv_bigint_outparam.phpt]: same for this one: remove the commented lines and unnecessary comments
also, you might want to do one as BIGINT and the other as VARCHAR (256 is unnecessarily large... you can try a smaller size for the VARCHAR)

File: test/functional/sqlsrv/sqlsrv_bigint_outparam.phpt [hunk 3]
```diff
@@ -25,15 +25,15 @@ AE\insertRow($conn, $tbname, array("c1_bigint" => 922337203685479936));
...
+$stmt = sqlsrv_prepare($conn, $outSql, array(array(&$bigintOut, SQLSRV_PARAM_OUT, null, SQLSRV_SQLTYPE_BIGINT)));
 sqlsrv_execute($stmt);
 printf("Large bigint output:\n");
 var_dump($bigintOut);
 printf("\n");
 
 // Call stored procedure with SQLSRV_PARAM_INOUT
 $bigintOut = 0;
-$stmt = sqlsrv_prepare($conn, $outSql, array(array(&$bigintOut, SQLSRV_PARAM_INOUT)));
+$stmt = sqlsrv_prepare($conn, $outSql, array(array(&$bigintOut, SQLSRV_PARAM_INOUT, null, SQLSRV_SQLTYPE_VARCHAR(20))));
```
Comment (yitam) [inline test/functional/sqlsrv/sqlsrv_bigint_outparam.phpt]: Having size = 20 might be a bit too small if it's a negative number (of course in this example it's fine)

File: test/functional/pdo_sqlsrv/pdo_bigint_outparam.phpt [hunk 4]
```diff
@@ -35,7 +35,7 @@ printf("\n");
 // Call stored procedure with inout
 $bigintOut = 0;
 $stmt = $conn->prepare($outSql);
-$stmt->bindParam(1, $bigintOut, PDO::PARAM_INT | PDO::PARAM_INPUT_OUTPUT, PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE);
+$stmt->bindParam(1, $bigintOut, PDO::PARAM_STR | PDO::PARAM_INPUT_OUTPUT, 2048);
 $stmt->execute();
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_bigint_outparam.phpt]: Perhaps not 2048 but a size that's a bit larger than 20?

File: test/functional/pdo_sqlsrv/pdo_ae_output_param_binary_size.phpt [hunk 5]
```diff
@@ -127,12 +141,8 @@ function testOutputBinary($inout)
...
                             }
                         } else {
-                            // $pdoParamType is PDO::PARAM_INT
-                            // This only occurs without AE -- likely a rare use case
-                            // With AE enabled, this would have caused an exception
-                            if (strval($det) != $ord0 || strval($rand) != $ord1) {
-                                printValues($errMsg, $det, $rand, $ord0, $ord1);
-                            }
+                            echo "Something went wrong. This should not have happened\n";
+                            printValues($errMsg, $det, $rand, $ord0, $ord1);
```
[the remark this answers is not in the document]
Comment (yitam) [reply test/functional/pdo_sqlsrv/pdo_ae_output_param_binary_size.phpt]: There are only five different PDO PARAM types, PARAM_NULL and PARAM_LOB will result in an exception, and now PARAM_INT will be skipped with non-AE. With AE, PARAM_INT will also result in an exception.

[the remark this answers is not in the document]
Comment (david-puglielli) [reply test/functional/pdo_sqlsrv/pdo_ae_output_param_binary_size.phpt]: ok I see

**Comment by david-puglielli** [comment]
Nope, I'll merge when AppVeyor completes successfully.

**Pull request merged**